tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/TraceAnnotationsTests.cs

```diff
@@ -86,7 +86,7 @@ public async Task SubmitTraces()
             ddTraceMethodsString += ";Samples.TraceAnnotations.ExtensionMethods[ExtensionMethodForTestType,ExtensionMethodForTestTypeGeneric,ExtensionMethodForTestTypeTypeStruct];System.Net.Http.HttpRequestMessage[set_Method]";
 
             SetEnvironmentVariable("DD_TRACE_METHODS", ddTraceMethodsString);
-
+            SetEnvironmentVariable("DD_TRACE_DEBUG", "1");
```

Comment:
Late comment since this is merged, but would it be better to test with _both_ debug mode enabled and disabled? With this change, if a bug is introduced that only occurs when debug mode is _disabled_, we wouldn't catch it, similar to how we didn't catch this bug that only happened when debug mode is enabled.

Reply:
+1

Reply:
Yep, totally agree on this - I didn't intend for this to be the final word on this, but rather just as a quick way to make sure this fix goes red-to-green.

I think this is something we probably should address in the context of our entire CI and not just this particular test - supposedly wacky problems such as this one could manifest in _any_ test, so it'd make sense to run our entire CI with both DD_TRACE_DEBUG and without. Then again, I suppose running _with_ it is problematic logistically (the size of the log artifacts would explode), so perhaps we want to do it as a scheduled CI build?

(I'll add this as a discussion topic for our bi-weekly meeting)